src/hb-ot-name-table.hh

```diff
@@ -113,6 +114,32 @@ struct name
     return_trace (true);
   }
 
+  inline unsigned int get_count (void) const
+  { return count; }
+
+  inline const NameRecord& get_record(unsigned int index) const
+  { return index < count ? nameRecordZ[index] : Null (NameRecord); }
```

Comment:
We actually have a new idiom for these.  We bind them to an `hb_array_t` that has size, then index:
`{ return hb_array_t<NameRecord> (nameRecordZ, count)[index]; }`